optimade/server/exception_handlers.py

```diff
@@ -37,21 +37,27 @@ def general_exception(
             Error(detail=detail, status=status_code, title=str(exc.__class__.__name__))
         ]
 
-    return JSONResponse(
-        status_code=status_code,
-        content=jsonable_encoder(
-            ErrorResponse(
-                meta=meta_values(
-                    url=str(request.url),
-                    data_returned=0,
-                    data_available=0,
-                    more_data_available=False,
-                    **{CONFIG.provider["prefix"] + "traceback": tb},
-                ),
-                errors=errors,
+    try:
+        response = ErrorResponse(
+            meta=meta_values(
+                url=str(request.url),
+                data_returned=0,
+                data_available=0,
+                more_data_available=False,
+                **{CONFIG.provider["prefix"] + "traceback": tb},
             ),
-            exclude_unset=True,
-        ),
+            errors=errors,
+        )
+    except Exception:
+        # This was introduced due to the original raise of an HTTPException if the
+        # path prefix could not be found, e.g., `/optimade/v0`.
+        # However, due to the testing, this error cannot be raised anymore.
+        # Instead, an OPTiMaDe warning should be issued.
+        # Having this try and except is still good practice though.
```

Comment:
Yeah ... not much at this moment in time - but I hope to implement the OPTiMaDe warnings similarly to Python `Warning`s, which are a sub-class of `Exception`s. I.e., at some point again (hopefully soon) there _will_ be raised something, which can be caught here and relayed as an OPTiMaDe `warning` ... But at the moment... yeah, not much.

Reply:
I don't quite follow - the `except` block here would only be activated if there was a programming error in the `try` block, i.e. in the preparation of the error response object.
In this case, one should not raise a Warning but an Error (since something would have gone seriously wrong).

The only difference I see between the response in `try` and in `except` is that the one in `except` does not add all the optimade stuff.
Perhaps one can simply remove the try/except here?

Reply:
The `except` block would be activated if the utility function `meta_values` raises, for one reason or another. Within `meta_values` is where the `meta`->`query`->`representation` is created and returned, which is why i originally created the `optional_base_urls` function - to return all 6 different path prefixes so they could be removed from the `urlparse`d path.

I don't know if this makes sense ... but if it should happen that the URL path of the server, for any reason, does not have the prefixes we set for it (e.g., `/optimade/v1`, which actually happens for the `TestClient` used to test the server) it would in my original commit raise a `HTTPException`. I tried to fix this in the test setup, but it eluded me, so instead I chose to backtrack, remove the `HTTPException` raised in `meta_values` and simply keep the `try` and `except` in this part - although it technically necessary (unless `meta_values` should fail due to other unknown reasons).